src/Refactor/EntityWriter/ProductXmlWriter.php

```diff
@@ -0,0 +1,238 @@
+     * Sets never export a brand
+     *
+     */
+    private function writeBrand(): void
+    {
+        if (! XmlFormatter::isEmptyValue($this->product->brand)) {
+            $this->writer->writeElement('brand', $this->product->brand);
+        } else {
+            if (! is_null($this->product->brand)) {
+                $this->writer->writeEmptyElement('brand');
+            }
+        }
+    }
+
+    private function writeImages(): void
+    {
+        $images = XmlFormatter::filterEmptyValues($this->product->images);
+
+        if (count($images) === 0) {
+            return;
+        }
+
+        $this->writer->startElement('images');
+        $this->writer->startElement('image-group');
+        $this->writer->writeAttribute('view-type', 'large');
+
+        foreach ($images as $image) {
+            $this->writer->ifNotEmpty()->writeEmptyElementWithAttributes('image', ['path' => $image]);
+        }
+
+        $this->writer->endElement();
+        $this->writer->endElement();
+    }
+
+    private function writePageAttributes(): void
+    {
+        /*
+        $pageAttributes = XmlFormatter::filterEmptyValues($this->product->pageAttributes);
+        */
+
+        $pageAttributes = $this->product->pageAttributes;
+
+        if (count($pageAttributes) === 0) {
+            return;
+        }
+
+        $this->writer->startElement('page-attributes');
+
+        foreach ($pageAttributes as $elemName => $elemContent) {
+            if (! XmlFormatter::isEmptyValue($elemContent)) {
+                $this->writer->writeElementWithAttributes($elemName, $elemContent, ['xml:lang' => 'x-default']);
+            } else {
+                $this->writer->writeEmptyElementWithAttributes($elemName, ['xml:lang' => 'x-default']);
+            }
+        }
+
+        $this->writer->endElement();
+    }
+
+    private function writeCustomAttributes(): void
+    {
+        if (count($this->product->customAttributes) === 0) {
+            return;
+        }
+
+        ksort($this->product->customAttributes);
+
+        $this->writer->startElement('custom-attributes');
+
+        foreach ($this->product->customAttributes as $customAttribute) {
+            $this->writer->writeEntity($customAttribute);
+        }
+
+        $this->writer->endElement();
+    }
+
+    private function writeVariations(): void
+    {
+        $attributes = XmlFormatter::filterEmptyValues($this->product->sharedVariationAttributes);
+        $variants   = XmlFormatter::filterEmptyValues($this->product->variants);
+
+        if (count($attributes) === 0 && count($variants) === 0) {
+            return;
+        }
+
+        $this->writer->startElement('variations');
+        $this->writeSharedVariationAttributes();
+        $this->writeVariants();
+        $this->writer->endElement();
+    }
+
+    private function writeSharedVariationAttributes(): void
+    {
+        $attributes = XmlFormatter::filterEmptyValues($this->product->sharedVariationAttributes);
```

Comment:
Should we pass this into this function from `writeVariations()` rather than doing it twice?

Reply:
@WillJW Good idea. Sorted, ta.